apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/entity/repository/WaUiMetadataRepository.java

```diff
@@ -46,17 +38,12 @@ Long countByPageAndQuestionIdentifier(
 
     @Query("Select COALESCE(Max(ui.orderNbr),0) FROM WaUiMetadata ui WHERE ui.waTemplateUid.id =:pageId")
     Integer findMaxOrderNbrForPage(@Param("pageId") Long pageId);
+    
+    public void deleteByWaTemplateUid(WaTemplate page);
 
-    public void deleteAllByWaTemplateUid(WaTemplate template);
+    @Query(value = QueryStore.FIND_PAGE_METADATA_BY_TEMPLATE_SQL,nativeQuery = true)
+    List<Object[]> findPageMetadataByWaTemplateUid(@Param("waTemplateUid") Long waTemplateUid);
```

Comment:
Recommendation: I would rather not pass this giant SQL query through the repository object. For this specific use case i think leveraging the `JdbcTemplate` makes sense. you can see the [PatientSummaryHomeAddressFinder](https://github.com/CDCgov/NEDSS-Modernization/blob/bc0286c8cfb85b35c9f38daa8a2da1f178385b13/apps/modernization-api/src/main/java/gov/cdc/nbs/patient/profile/summary/address/PatientSummaryHomeAddressFinder.java#L14) for an example.